docs/source/user/settings.rst

```diff
@@ -314,6 +314,32 @@ the configuration should be simply changed to:
         'default': 'https://login.wifiservice.com/{organization}/password/reset/confirm/{uid}/{token}'
     }
 
+
+``OPENWISP_RADIUS_OPTIONAL_REGISTRATION_FIELDS``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Default**: ``{'first_name': 'disabled', 'last_name': 'disabled', 'location': 'disabled', 'birth_date': 'disabled'}``
+
+This setting is used to specify if the user's optional fields (``first_name``, ``last_name``, ``location`` and ``birth_date``) 
+should either be optionally allowed, ignored or required in the `User Registration API <api.html#user-registration>`_.
+It can take any of the following values:
+
+- ``allowed``: value of user's optional fields will be save to the database but they are not required.
+- ``disabled``: value of user's optional fields are not required and will not be saved to the database if supplied.
+- ``mandatory``: value of user's optional fields are required and will be saved to the database.
+
+For example:
+
+.. code-block:: python
+
+    OPENWISP_RADIUS_OPTIONAL_REGISTRATION_FIELDS = {'first_name': 'disabled', 'last_name': 'disabled', 'location': 'allowed', 'birth_date': 'mandatory'}
+
+In the example above;
+
+- ``first_name`` and ``last_name`` fields are not required and their values if provided are ignored.
+- ``location`` field is not required but its value will be save to the database if provided.
+- ``birth_date`` field is required and a ``ValidationError`` exception is raised if its value is not provided. 
+
```

Comment:
please update this page as well: https://openwisp-radius.readthedocs.io/en/latest/user/api.html#user-registration

Mentioning the new fields, mentioning the fields are disabled by default and can be enabled (link this setting).